src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UseStringContainsCharOverloadWithSingleCharactersTests.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.UseStringContainsCharOverloadWithSingleCharactersAnalyzer,
+    Microsoft.NetCore.Analyzers.Performance.UseStringContainsCharOverloadWithSingleCharactersCodeFix>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.UseStringContainsCharOverloadWithSingleCharactersAnalyzer,
+    Microsoft.NetCore.Analyzers.Performance.UseStringContainsCharOverloadWithSingleCharactersCodeFix>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class UseStringContainsCharOverloadWithSingleCharactersTests
+    {
+        [Fact]
+        public async Task CSharp_RegularStringLiteralWithoutCharOverloadAvailable_NoDiagnostic()
+        {
+            var sourceCode = @"using System;
+public class TestClass
+{
+    public bool TestMethod(string input)
+    {
+        return ""testString"".Contains({|#0:""a""|});
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(sourceCode, ReferenceAssemblies.NetStandard.NetStandard20, DiagnosticResult.EmptyDiagnosticResults);
+        }
+
+        [Fact]
+        public async Task CSharp_RegularStringLiteral_Diagnostic()
+        {
+            var violatingSourceCode = @"using System;
+public class TestClass
+{
+    public bool TestMethod(string input)
+    {
+        return ""testString"".Contains({|#0:""a""|});
+    }
+}";
+            var expectedDiagnostic = VerifyCS.Diagnostic(UseStringContainsCharOverloadWithSingleCharactersAnalyzer.s_rule_CA1839)
+                .WithLocation(0);
+            await VerifyCS.VerifyAnalyzerAsync(violatingSourceCode, ReferenceAssemblies.NetStandard.NetStandard21, expectedDiagnostic);
```

Comment:
Any tests with expected diagnostics should use VerifyCodeFixAsync

Reply:
> Any tests with expected diagnostics should use VerifyCodeFixAsync.

Yes, specifically for the reason that all calls to `VerifyCodeFixAsync` _automatically_ call `VerifyAnalyzerAsync`, so it's literally twice the work for the same result.

Reply:
Its just a relic of the process. I wrote tests to verify my analyzer first and then moved on with the codefixer, but you're right it looks pretty obsolete given the other tests.